### PR TITLE
Changes done to make sure , schemas String is visible in case of the …

### PR DESCRIPTION
**Pull request by rohitsoni77**
Errors, currently it gets wrapped into a String array and the object representation is being shown when JSONObject.toString() is called , fixed the code to correct the representation of the string , by creating a JSONArray of Strings and adding the same to the JSONObject.

Errors, currently it gets wrapped into a String array and the object representation is being shown when JSONObject.toString() is called

Fixed the code to correct the representation of the string , by creating a JSONArray of Strings and adding the same to the JSONObject.

Fixed the code to correct the representation of the string , by creating a JSONArray of Strings and adding the same to the JSONObject.

## Automation tests
 - Unit tests 
   > Code coverage information
 - Integration tests
   > Details about the test cases and coverage

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes/no
 - Ran FindSecurityBugs plugin and verified report? yes/no
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes/no

### CODE REVIEW

**Comment by tharindu-bandara** [comment]
Hi @rohitsoni77 ,

Thank you very much for the contribution! 

However, it seems like the branch which this PR is made in to(3.0.x) is no longer the active development branch.

Let's make this PR to the [active development branch[1]](https://github.com/wso2/charon/tree/master) to push it further through the review process and get merged. 

PS: Before you start on this, kindly checkout to the active development branch[1] and apply your fix on top of that. That way the commits would look much cleaner, and easy to review.

[1] https://github.com/wso2/charon/tree/master

File: modules/charon-core/src/main/java/org/wso2/charon3/core/encoder/JSONEncoder.java [hunk 1]
```diff
@@ -97,7 +98,8 @@ public String encodeSCIMException(AbstractCharonException exception) {
 
         try {
             //construct error object with details in the exception
-            errorObject.put(ResponseCodeConstants.SCHEMAS, new String[]{exception.getSchemas()});
+            JSONArray arrSchemas = new JSONArray(Arrays.asList(new String[]{exception.getSchemas()}));
```
Comment (tharindu-bandara) [inline modules/charon-core/src/main/java/org/wso2/charon3/core/encoder/JSONEncoder.java]:
```suggestion
            JSONArray arrSchemas = new JSONArray(Collections.singletonList(exception.getSchemas()));
```
Shall we simplify the array generation?

**Comment by tharindu-bandara** [comment]
@rohitsoni77 

Kindly sign the contributor license agreement, so that we can merge this PR once the above comment is addressed.

**Comment by rohitsoni77** [comment]
@tharindu-bandara - The first commit in this branch was made by the "root" user , and there is no corresponding git account to that user and hence cla not signed for that user , subsequent commits are made by user rohitsoni77 , and I have signed cla using that , can you please suggest how to resolve the issue for the "root" user.

**Comment by tharindu-bandara** [comment]
> @tharindu-bandara - The first commit in this branch was made by the "root" user , and there is no corresponding git account to that user and hence cla not signed for that user , subsequent commits are made by user rohitsoni77 , and I have signed cla using that , can you please suggest how to resolve the issue for the "root" user.

I believe without the corresponding GitHub account, signing the CLA may not possible :( . But as you have already signed the CLA, would it be possible for you to do one of the following?

1. soft reset the git commits in this PR(so that `root` user's record is removed), and commit again to make a single commit under your name(rohitsoni77) containing this fix.
2. Create a new PR and make a commit under your name(rohitsoni77) with the changes done in this PR and get it merged, then close this PR.

Let us know your feedback. As this pull request has already gone through the PR process, we can merge any of the outcomes from above.

Please accept my sincere apologies as you had to do several changes regarding this PR. I hope you would understand the requirement of signing the CLA, and help us to get this PR merged.

Tharindu.

**Comment by tharindu-bandara** [comment]
Hi @rohitsoni77 ,

Looks like the root user's commit is still there as the CLA pending verification still persists.

Would it be possible for you to create a fresh pull request instead of this(as the fix has a two lines change)? If possible, then kindly make sure only to commit as the user, `rohitsoni77`.

Please let me know your feedback.

Tharindu.

**Comment by rohitsoni77** [comment]
@tharindu-bandara - I have created new pull request 
 with same changes - rohitsoni77:rosoni_fix_errors_schemas_representation , Please approve that merge

**Pull request closed without merging**